### PR TITLE
Fill in benchmark results workflow

### PR DESCRIPTION
**Pull request by dbarbuzzi**
SUMMARY:
This PR adds a workflow that is triggered by external repository, given information to download benchmark results, and then processes those results/updates the repo’s pages branch `nm-gh-pages` branch using the same approach the previous, public `nm-vllm` repo did.

TEST PLAN:
* External triggering was tested with a stub workflow: https://github.com/neuralmagic/nm-vllm-certs/actions/runs/10045049642
* Isolated testing of this workflow with hard-coded values was tested separately:
  * Run: https://github.com/neuralmagic/nm-vllm-certs/actions/runs/10062146301
  *  Updated branch: `nm-gh-pages-temp` (will be deleted after merge)

### CODE REVIEW

File: .github/workflows/process-benchmark-results.yaml [hunk 1]
```diff
@@ -6,11 +6,88 @@ on:
...
+            project_id: ${{ secrets.GCP_PROJECT }}
+            workload_identity_provider: ${{ secrets.GCP_WORKLOAD_IDENTITY_PROVIDER }}
+            service_account: ${{ secrets.GCP_GHA_SA }}
+
+      - name: Set up Google Cloud SDK
+        uses: google-github-actions/setup-gcloud@v2
+        with:
+            version: '>= 473.0.0'
+
+      - name: Download GHA-formatted JSON files
```
Comment (andy-neuma) [inline .github/workflows/process-benchmark-results.yaml:43]: we need to obfuscate the bucket and asset layout.

**Review by andy-neuma — COMMENTED** [review]
thanks

File: .github/workflows/process-benchmark-results.yaml (hunk 1 above)
Comment (dbarbuzzi) [reply .github/workflows/process-benchmark-results.yaml:43]: Good catch; this should be resolved now.

**Review by andy-neuma — APPROVED** [review]
thanks

**Pull request merged**